src/block/block_ckpt.c

```diff
@@ -722,6 +782,15 @@ __ckpt_update(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_CKPT *ckptbase, WT_C
         WT_RET(ret);
     }
 
+    /*
+     * If this is the live system, we need to record the list of blocks written for this checkpoint
+     * (including the blocks we allocated to write the extent lists).
+     */
+    if (F_ISSET(ckpt, WT_CKPT_BLOCK_MODS)) {
+        WT_ASSERT(session, is_live == true);
+        WT_RET(__ckpt_add_blk_mods(session, ckpt, ci));
+    }
```

Comment:
I'd assert this in `__ckpt_verify()`, there are similar checks there.

Reply:
Done.